ttnn/cpp/ttnn/operations/moreh/moreh_adam/device/moreh_adam_device_operation.hpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <variant>
+#include <vector>
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_adam {
+struct MorehAdamOperation {
+    struct operation_attributes_t {
```

Comment:
Thanks for the review. My thought here was that since the only way you could ever construct it is through `invoke()`, which will guarantee that these exists, it wouldn’t be an issue. I’ll make the changes necessary to follow the best practices docs

Reply:
I've added the default values. Please check if the code is as you expect.